src/Payments/Payment.php

```diff
@@ -0,0 +1,487 @@
+     * @JMS\XmlElement(cdata=false)
+     * @JMS\Groups({"request", "response"})
+     */
+    private $paymentType;
+
+    /**
+     * @var string
+     * @JMS\Type("string")
+     * @JMS\SerializedName("account")
+     * @JMS\Groups({"request", "response"})
+     */
+    private $account;
+
+    /**
+     * @var float
+     * @JMS\Type("double")
+     * @JMS\SerializedName("value_pln")
+     * @JMS\Groups({"request", "response"})
+     */
+    private $valuePln;
+
+    /**
+     * @var float
+     * @JMS\Type("double")
+     * @JMS\SerializedName("currency_exchange")
+     * @JMS\Groups({"response"})
+     */
+    private $currencyExchange;
+
+    /**
+     * @var string
+     * @JMS\Type("string")
+     * @JMS\SerializedName("currency_label")
+     * @JMS\Groups({"response"})
+     */
+    private $currencyLabel;
+
+    /**
+     * @var \DateTime
+     * @JMS\Type("DateTime<'Y-m-d'>")
+     * @JMS\SerializedName("currency_date")
+     * @JMS\Groups({"response"})
+     */
+    private $currency_date;
+
+    /**
+     * @var int
+     * @JMS\Type("integer")
+     * @JMS\SerializedName("notes")
+     * @JMS\XmlElement(cdata=false)
+     * @JMS\Groups({"request", "response"})
+     */
+    private $notes;
+
+    /**
+     * @var TagIds
+     * @JMS\Type("Webit\WFirmaSDK\Tags\TagIds")
+     * @JMS\SerializedName("tags")
+     * @JMS\Groups({"request", "response"})
+     */
+    private $tags;
+
+    /**
+     * @var InvoiceId
+     * @JMS\Type("Webit\WFirmaSDK\Invoices\InvoiceId")
+     * @JMS\SerializedName("invoice")
+     * @JMS\XmlElement(cdata=false)
+     * @JMS\Groups({"response"})
+     */
+    private $invoice;
+
+    /**
+     * @var ExpenseId
+     * @JMS\Type("Webit\WFirmaSDK\Expenses\ExpenseId")
+     * @JMS\SerializedName("expense")
+     * @JMS\XmlElement(cdata=false)
+     * @JMS\Groups({"response"})
+     */
+    private $expense;
+
+    /**
+     * @var SeriesId
+     * @JMS\Type("Webit\WFirmaSDK\Series\SeriesId")
+     * @JMS\SerializedName("series")
+     * @JMS\XmlElement(cdata=false)
+     * @JMS\Groups({"response"})
+     */
+    private $series;
+
+    /**
+     * @var PaymentCashboxId
+     * @JMS\Type("Webit\WFirmaSDK\PaymentCashboxes\PaymentCashboxId")
+     * @JMS\SerializedName("payment_cashbox")
+     * @JMS\XmlElement(cdata=false)
+     * @JMS\Groups({"response"})
+     */
+    private $paymentCashbox;
+
+    /**
+     * @var ContractorId
+     * @JMS\Type("Webit\WFirmaSDK\Contractors\ContractorId")
+     * @JMS\SerializedName("contractor")
+     * @JMS\XmlElement(cdata=false)
+     * @JMS\Groups({"response"})
+     */
+    private $contractor;
+
+    /**
+     * Payment constructor.
+     * @param Invoice $invoice
+     * @param float $value
+     * @param \DateTimeInterface|null $date
+     * @param PaymentMethod|null $paymentMethod
+     * @param PaymentCashboxId|null $cashboxId
+     * @param TagIds|null $tags
+     */
+    private function __construct(
+        $object,
```

Comment:
Konstruktor powinien przyjmować nie obiekt jako taki, ale typ obiektu (tak jak w dokumentacji API). Konstruktory statyczne (from*) powinny przekazywać poprawny typ obiektu do konstruktora.

Reply:
Czyli chcesz $object zmienić na dwa parametry: $objectName, $objectId (bo tak to wygląda w API)?

Reply:
Tak sądzę. Jeśli chodzi o te obiekty, to ich atrybuty zwykle mapuje 1:1 z atrybutami API. Abstrakcję (grupowanie kilku atrybutów dla wygody) dodaję tylko w getterach / setterach i konstruktorze.

Reply:
Zmienione.